Light, Shape, Messages, Material Params - interacting with scene mess…

tcadanklefsen opened the pull request:
…ages.

- Refactor out copying/saving references to custommodelviewer everywhere, turned into singleton (for now). Important info to be extracted out to new singleton. 
- Shape manager v1 boilerplate in with a 2 sided cube. Tasking to add more shapes backlogged. 
- Model loader now loads multiple models.
- Material params are now set (color and float only added properties for now), with warning if you try to set materials not declared  yet. 
- Added ability to dynamically change lighting easier. 
- Added messages for more gameplay related / demo-ness.

tcadanklefsen committed:
Light, Shape, Messages, Material Params - interacting with scene messages.

Refactor out copying/saving references to custommodelviewer everywhere, turned into singleton (for now). Important info to be extracted out to new singleton.
Shape manager v1 boilerplate in with a 2 sided cube. Tasking to add more shapes backlogged.
Model loader now loads multiple models.
Material params are now set (color and float only added properties for now), with warning if you try to set materials not declared  yet.
Added ability to dynamically change lighting easier.
Added messages for more gameplay related / demo-ness.

tcadanklefsen committed:
Light, Shape, Messages, Material Params - interacting with scene messages.

Refactor out copying/saving references to custommodelviewer everywhere, turned into singleton (for now). Important info to be extracted out to new singleton.
Shape manager v1 boilerplate in with a 2 sided cube. Tasking to add more shapes backlogged.
Model loader now loads multiple models.
Material params are now set (color and float only added properties for now), with warning if you try to set materials not declared  yet.
Added ability to dynamically change lighting easier.
Added messages for more gameplay related / demo-ness.

jwinarske commented on plugins/filament_view/core/include/file_utils.h:
```diff
@@ -83,14 +83,14 @@ inline std::vector<uint8_t> readBinaryFile(const std::string& dependent_path,
   }
 
-  SPDLOG_DEBUG("Reading: {}/{}", main_path, dependent_path);
+  SPDLOG_TRACE("Reading: {}/{}", main_path, dependent_path);
```
This should remain debug.  Trace enabled includes the backend context switching.  It get's very busy.

jwinarske commented on plugins/filament_view/core/include/file_utils.h:
```diff
@@ -83,14 +83,14 @@ inline std::vector<uint8_t> readBinaryFile(const std::string& dependent_path,
   const auto size = static_cast<std::size_t>(end - file.tellg());
 
-  SPDLOG_DEBUG("Read: {} bytes", size);
+  SPDLOG_TRACE("Read: {} bytes", size);
```
same as above.

jwinarske commented on plugins/filament_view/viewer/custom_model_viewer.h:
```diff
@@ -205,6 +214,11 @@ class CustomModelViewer {
+
+  // elapsed time needs to be moved to its own global namespace like class
+  // similar to unitys, elapsedtime/total time etc.
+  void doDemoGameplayLoop(const float fDeltaTime);
```
what is this for?

jwinarske commented on plugins/filament_view/viewer/custom_model_viewer.cc:
```diff
@@ -210,15 +237,36 @@ void CustomModelViewer::setupView() {
+      //
+      // Future tasking for making a more featured timing / frame info class.
+      uint32_t deltaTime = time - G_LastTime;
+      doDemoGameplayLoop((float)deltaTime / 1000);
```
Please add anything related to a demo under a #define, with it disabled by default.

jwinarske reviewed (changes requested):
* Trace is only used for critical bring up items
* All uppercase are only present in Debug builds, and get optimized out in Release builds.
* So either spdlog::debug (to include in Release), or SPDLOG_DEBUG
* Demo features need to be under a #define flag, defaulting to off.  Either define option in local CMake, or as #define in top level to library

tcadanklefsen committed:
Went from trace to info; set errors if failure to load on files.

tcadanklefsen committed:
Went from trace to info; set errors if failure to load on files.

tcadanklefsen committed:
Removed demo code. Added camera features, an autoorbit being a feature. Camera manager now knows about a primary camera (following other engine paradigms).

tcadanklefsen committed:
Removed demo code. Added camera features, an autoorbit being a feature. Camera manager now knows about a primary camera (following other engine paradigms).

tcadanklefsen committed:
Misc file cleanup, moving initialization to private.

tcadanklefsen committed:
Misc file cleanup, moving initialization to private.

jwinarske committed:
clang-tidy and clang-format files

Signed-off-by: Joel Winarske <joel.winarske@toyotaconnected.com>

jwinarske committed:
Merge pull request #4 from toyota-connected/jw/clang-tidy-format

clang-tidy and clang-format files

tcadanklefsen committed:
Adding logging to subsurface creation.

tcadanklefsen committed:
Adding logging to subsurface creation.

tcadanklefsen committed:
Added a wait() before moving pointers, prevents segfault

tcadanklefsen committed:
Added a wait() before moving pointers, prevents segfault

tcadanklefsen committed:
Added a move to fix pointer issue. Cleaned up movement code.

tcadanklefsen committed:
Added a move to fix pointer issue. Cleaned up movement code.

tcadanklefsen committed:
Default camera values added.

tcadanklefsen committed:
Default camera values added.

tcadanklefsen committed:
Checking in clang items.

tcadanklefsen committed:
Checking in clang items.

tcadanklefsen committed:
Clang formatted plugins files.

tcadanklefsen committed:
Clang formatted plugins files.

tcadanklefsen committed:
Clang Format CC files and clang Tidy H files. Removed unused classes/files.

tcadanklefsen committed:
Clang Format CC files and clang Tidy H files. Removed unused classes/files.

tcadanklefsen committed:
Tidy and format updates.

tcadanklefsen committed:
Tidy and format updates.

tcadanklefsen committed:
Runtime fix for not rotating camera unless rotate camera (auto) was unlocked.

tcadanklefsen committed:
Runtime fix for not rotating camera unless rotate camera (auto) was unlocked.

tcadanklefsen committed:
Removing unused variable.

tcadanklefsen committed:
Removing unused variable.

tcadanklefsen committed:
No longer need to build mvp

tcadanklefsen committed:
No longer need to build mvp

jwinarske commented on plugins/filament_view/CMakeLists.txt:
```diff
@@ -130,22 +130,22 @@ target_link_libraries(plugin_filament_view PUBLIC
-        target_link_options(filament-mvp PRIVATE $<$<COMPILE_LANGUAGE:CXX>:-fuse-ld=lld -l:libc++.a -l:libc++abi.a -static-libgcc -lc -lm -v>)
-    endif ()
-    add_sanitizers(filament-mvp)
+# pkg_check_modules(SDL2 IMPORTED_TARGET sdl2)
```
Why is this commented out?

tcadanklefsen replied:
This one fails to build by default.

jwinarske replied:
fine for now.  will address in a later PR

jwinarske commented on plugins/filament_view/core/model/loader/model_loader.cc:
```diff
@@ -124,16 +153,20 @@ void ModelLoader::loadModelGltf(
+  auto* asset = assetLoader_->createAsset(buffer.data(),
+                                          static_cast<uint32_t>(buffer.size()));
+  if (!asset) {
+    // TODO THROW ERROR
```
Agreed.  It needs an error class

jwinarske commented on plugins/filament_view/core/model/loader/model_loader.h:
```diff
@@ -92,47 +89,47 @@ class ModelLoader {
-  void fetchResources(
-      ::filament::gltfio::FilamentAsset* asset,
-      const std::function<uint8_t*(std::string asset)> callback);
+  //   void fetchResources(
```
best to use 
```
#if 0 //TODO
...
#endif  //TODO
```

tcadanklefsen replied:
Leftover from previous, removed.

jwinarske commented on plugins/filament_view/core/scene/camera/camera.cc:
```diff
@@ -147,18 +154,23 @@ Camera::Camera(const flutter::EncodableMap& params) {
-        orbitHomePosition_ =
-            std::make_unique<::filament::math::float3>(0, 0, 1);
-      }
+      }  // else if (std::holds_alternative<std::monostate>(it.second)) {
```
use #if 0 pattern as mentioned above

jwinarske commented on plugins/filament_view/core/scene/camera/camera.h:
```diff
@@ -53,12 +53,18 @@ class Camera {
   static constexpr char kModeFreeFlight[] = "FREE_FLIGHT";
   static constexpr char kFovVertical[] = "VERTICAL";
   static constexpr char kFovHorizontal[] = "HORIZONTAL";
+  static constexpr char kModeAutoOrbit[] = "AUTOORBIT";
```
Can you document AUTOORBIT somewhere?

tcadanklefsen replied:
There's additional tasking backlogged for improvement here in another iteration; but yes, adding more info for now.

tcadanklefsen replied:
Added comments, re-review

jwinarske commented on plugins/filament_view/core/scene/camera/camera_manager.cc:
```diff
@@ -270,7 +286,9 @@ void CameraManager::updateCameraManipulator(Camera* cameraInfo) {
   }
 
-  auto viewport = modelViewer_->getFilamentView()->getViewport();
+  CustomModelViewer* modelViewer = CustomModelViewer::Instance(__FUNCTION__);
```
How would this work with multiple filament views?

jwinarske replied:
Will be addressed in later PR

jwinarske commented on plugins/filament_view/core/scene/material/loader/material_loader.cc:
```diff
@@ -37,14 +42,42 @@ Resource<::filament::Material*> MaterialLoader::loadMaterialFromUrl(
+
+void MaterialLoader::PrintMaterialInformation(
+    const ::filament::Material* material) {
+  SPDLOG_DEBUG("Material Informaton {}", material->getName());
```
These should be spdlog::info.  Say you want to print material info in release.  Calling this would produce no output.

jwinarske commented on plugins/filament_view/core/scene/material/loader/texture_loader.cc:
```diff
@@ -29,18 +23,24 @@ inline ::filament::backend::TextureFormat internalFormat(
     Texture::TextureType type,
     std::unique_ptr<image::LinearImage> image) {
-  if (image->getChannels() != 3) {
+  static const int NUM_CHANNELS_FOR_IMAGE = 3;
```
constexpr, and use `kNumChannelsForImage` naming pattern.

tcadanklefsen replied:
Fixed.

jwinarske commented on plugins/filament_view/core/scene/material/loader/texture_loader.cc:
```diff
@@ -97,17 +102,9 @@ ::filament::Texture* TextureLoader::loadTextureFromStream(
 }
 
 ::filament::Texture* TextureLoader::loadTextureFromUrl(
-    std::string url,
```
We can't remove network loading of assets.  Consider a development workflow.

tcadanklefsen replied:
Adding this back in, this was failing to build on my fedora instance.

jwinarske commented on plugins/filament_view/core/scene/scene_controller.cc:
```diff
@@ -75,21 +78,24 @@ void SceneController::setUpViewer(PlatformView* platformView,
-  f.wait();
+  groundManager_ = std::make_unique<GroundManager>(scene_->ground_.get());
+  groundManager_->createGround(materialManager_.get());
+  // auto f = groundManager_->createGround();
```
Use `#if 0 //TODO - comment` so we know why it's commented and not deleted

tcadanklefsen replied:
Removed / added comment.

jwinarske commented on plugins/filament_view/core/shapes/shape.cc:
```diff
@@ -16,53 +16,310 @@
+  return true;
+}
+
+#if 1
```
this should be removed

tcadanklefsen replied:
Added comment, next tasking to be implemented of moving init code out of here.

jwinarske commented on plugins/filament_view/core/shapes/shape.cc:
```diff
@@ -16,53 +16,310 @@
+  // There's a bug with this, where they *MIGHT* show up, but even then
+  // totally incorrectly. Keeping for next iteration.
+
+  // float unitVertices[] = {
```
use #if 0 //TODO - comment pattern, or delete

tcadanklefsen replied:
Active work for next iteration.

jwinarske commented on plugins/filament_view/core/shapes/shape.cc:
```diff
@@ -16,53 +16,310 @@
+      .build(*engine_, *m_poEntity);
+}
+#else
+void Shape::createCube(::filament::Engine* engine_,
```
#if 0 //TODO - comment pattern or delete

tcadanklefsen replied:
Added comment, keeping to split in next'ish PR

jwinarske commented on plugins/filament_view/messages.g.cc:
```diff
@@ -50,43 +52,120 @@ void FilamentViewApi::SetUp(flutter::BinaryMessenger* binary_messenger,
             if (methodCall.method_name() == "CHANGE_ANIMATION_BY_INDEX") {
               result->Success();
+            }
+            // TODO Make a literals for these values:
```
These should be added now using this pattern:
```
`static constexpr char kChangeLightColorByIndex[] = "CHANGE_LIGHT_COLOR_BY_INDEX"`
```

tcadanklefsen replied:
Fixed. Thanks

jwinarske commented on plugins/filament_view/messages.g.cc:
```diff
@@ -50,43 +52,120 @@ void FilamentViewApi::SetUp(flutter::BinaryMessenger* binary_messenger,
+
+              result->Success();
+            }
+            // const String _toggleShapesInScene = "TOGGLE_SHAPES_IN_SCENE";
```
same as above

tcadanklefsen replied:
Fixed.

jwinarske commented on plugins/filament_view/messages.g.cc:
```diff
@@ -50,43 +52,120 @@ void FilamentViewApi::SetUp(flutter::BinaryMessenger* binary_messenger,
+              }
+              result->Success();
+            }
+            // const String _toggleCameraAutoRotate =
```
same as above

tcadanklefsen replied:
Fixed.

jwinarske commented on plugins/filament_view/messages.g.cc:
```diff
@@ -50,43 +52,120 @@ void FilamentViewApi::SetUp(flutter::BinaryMessenger* binary_messenger,
             } else {
               result->NotImplemented();
             }
 #if 0
```
needs a //TODO - comment.  We need to know what's going on.  Should we delete it, or use it.  It adds technical debt

tcadanklefsen replied:
This was prior code, no idea why its there - didnt write it. But i can definitely delete it if you want?

jwinarske replied:
it needs to be deleted.  Was a working example that should have been removed

tcadanklefsen replied:
Deleted

jwinarske commented on plugins/filament_view/viewer/custom_model_viewer.h:
```diff
@@ -83,8 +81,8 @@ class CustomModelViewer {
   // Disallow copy and assign.
-  CustomModelViewer(const CustomModelViewer&) = delete;
-  CustomModelViewer& operator=(const CustomModelViewer&) = delete;
+  // CustomModelViewer(const CustomModelViewer&) = delete;
```
use #if 0 //TODO - comment, or delete

jwinarske replied:
Are you copying or assigning CustomModelViewer?

tcadanklefsen replied:
I am not, added back in.

jwinarske commented on plugins/nav_render_view/libnav_render.h:
this file is in an adjacent module.  It should only be limited to the filament view plugin

jwinarske commented on plugins/nav_render_view/nav_render_error.h:
same as above

jwinarske commented on plugins/nav_render_view/nav_render_surface.cc:
same as above

jwinarske commented on plugins/nav_render_view/nav_render_texture.cc:
same as above

jwinarske commented on plugins/nav_render_view/nav_render_texture.h:
same as above

jwinarske commented on clang-format:
this file is already checked in

jwinarske commented on clang-tidy:
this file is already checked in

jwinarske reviewed (changes requested):
I checked in the clang-format and clang-tidy when I mentioned it prior.  You'll need to update that and rebase.

* Comment using `#if 0 //TODO - <comment>` pattern.  It needs to indicate why it's there.  If not a good reason delete the code.
* A singleton for the model viewer breaks multiple filament views at the same time
* The nav render plugin should not be part of this PR

Another important note is that we need to be able to construct and destroy any number of filament views at runtime; the destruction path needs to be clean.

tcadanklefsen committed:
Reverting clanged tidy/format files.

tcadanklefsen committed:
Reverting clanged tidy/format files.

tcadanklefsen committed:
More adjacent file removals for tidy/format.

tcadanklefsen committed:
More adjacent file removals for tidy/format.

tcadanklefsen committed:
Removing format/tidy from outside filamentview/common.

tcadanklefsen committed:
Removing format/tidy from outside filamentview/common.

tcadanklefsen committed:
PR cleanup.

tcadanklefsen committed:
PR cleanup.

tcadanklefsen committed:
Const the literals for lookup.

tcadanklefsen committed:
Const the literals for lookup.

tcadanklefsen committed:
Compile fix.

tcadanklefsen committed:
Compile fix.

tcadanklefsen committed:
Constexpr change. Warning remove.

tcadanklefsen committed:
Constexpr change. Warning remove.

tcadanklefsen committed:
Added comment of intended use.

tcadanklefsen committed:
Added comment of intended use.

tcadanklefsen committed:
Added comments, macro to non-macro spdlog

tcadanklefsen committed:
Added comments, macro to non-macro spdlog

tcadanklefsen committed:
Added comments.

tcadanklefsen committed:
Added comments.

tcadanklefsen committed:
Added comment of intended next steps.

tcadanklefsen committed:
Added comment of intended next steps.

tcadanklefsen committed:
Adding code for loading texture from url.

tcadanklefsen committed:
Adding code for loading texture from url.

tcadanklefsen committed:
Removing files.

tcadanklefsen committed:
Removing files.

tcadanklefsen committed:
Added error messages on load attempts.

tcadanklefsen committed:
Added error messages on load attempts.

jwinarske committed:
baseline CI jobs

-ivi-homescreen
-clang-format
-clang-tidy

Signed-off-by: Joel Winarske <joel.winarske@toyotaconnected.com>

jwinarske committed:
SDL2 compatibility

-switch SDL_GetWindowSizeInPixels to SDL_GetWindowSize
-remove event wheel usage

Signed-off-by: Joel Winarske <joel.winarske@toyotaconnected.com>

jwinarske committed:
Merge pull request #6 from toyota-connected/jw/fw_mvp

switch SDL_GetWindowSizeInPixels to SDL_GetWindowSize

tcadanklefsen committed:
Removing unused example code.

tcadanklefsen committed:
Removing unused example code.

tcadanklefsen committed:
Added back in

tcadanklefsen committed:
Added back in

jwinarske committed:
Merge pull request #5 from toyota-connected/jw/ci

baseline CI job

tcadanklefsen committed:
Merge branch 'Mv2_Lights_Shapes_Messages' of github.com:tcadanklefsen/ivi-homescreen-plugins into Mv2_Lights_Shapes_Messages
